src-electron/db/db-mapping.js

```diff
@@ -475,36 +481,61 @@ exports.map = {
+      clusterRef: x.CLUSTER_REF,
+      clusterSide: x.SIDE,
+      code: x.CODE, // Attribute Code
+      defaultValue: x.DEFAULT_VALUE,
+      define: x.DEFINE, // Attribute define
+      endpointId: x.ENDPOINT_IDENTIFIER, // Endpoint type attribute's endpoint Id
+      endpointTypeRef: x.ENDPOINT_TYPE_REF,
+      entryType: x.ARRAY_TYPE,
+      hexCode: '0x' + bin.int16ToHex(x['CODE'] ? x['CODE'] : 0), // Attribute code in hex
+      id: x.ATTRIBUTE_ID, // Attribute id
+      included: dbApi.fromDbBool(x.INCLUDED),
+      includedReportable: dbApi.fromDbBool(x.INCLUDED_REPORTABLE), // Is attribute reportable
+      isArray: x.IS_ARRAY, // Is attribute of type array
+      isBound: dbApi.fromDbBool(x.BOUNDED), // Is endpoint type attribute bounded
+      isClusterEnabled: x.ENABLED,
+      isGlobalAttribute: x.IS_GLOBAL_ATTRIBUTE, // Is attribute global
+      isIncluded: dbApi.fromDbBool(x.INCLUDED), // Is endpoint type attribute included
       isManufacturingSpecific: dbApi.toDbBool(
         x.MANUFACTURER_CODE | x.CLUSTER_MANUFACTURER_CODE
       ), // Is Attribute mfg specific or not
-      endpointId: x.ENDPOINT_IDENTIFIER, // Endpoint type attribute's endpoint Id
-      tokenId: x.TOKEN_ID, // Endpoint type attribute's token id
+      isNullable: dbApi.fromDbBool(x.IS_NULLABLE), // Is attribute nullable
+      isOptionalAttribute: dbApi.fromDbBool(x.IS_OPTIONAL),
+      isReportableAttribute: dbApi.fromDbBool(x.INCLUDED_REPORTABLE), // Is attribute reportable
+      isSceneRequired: dbApi.fromDbBool(x.IS_SCENE_REQUIRED),
+      isSingleton: dbApi.fromDbBool(x.SINGLETON), // Endpoint type attribute is singleton or not
+      isWritable: dbApi.fromDbBool(x.IS_WRITABLE), // Is attribute writable
+      isWritableAttribute: dbApi.fromDbBool(x.IS_WRITABLE), // Is attribute writable
+      manufacturerCode: x.MANUFACTURER_CODE
+        ? x.MANUFACTURER_CODE
+        : x.CLUSTER_MANUFACTURER_CODE, // Attribute manufacturer code
+      max: x.MAX, // Attribute max value
+      maxInterval: x.MAX_INTERVAL,
+      maxLength: x.MAX_LENGTH, // Attribute max length
+      mfgCode: x.MANUFACTURER_CODE
+        ? x.MANUFACTURER_CODE
+        : x.CLUSTER_MANUFACTURER_CODE, // Attribute manufacturer code
+      min: x.MIN, // Attribute min value
+      minInterval: x.MIN_INTERVAL,
+      minLength: x.MIN_LENGTH, // Attribute min length
+      mustUseTimedWrite: dbApi.fromDbBool(x.MUST_USE_TIMED_WRITE),
+      name: x.NAME, // Attribute Name
+      reportableChange: x.REPORTABLE_CHANGE,
+      side: x.SIDE, // Attribute Side
+      singleton: dbApi.fromDbBool(x.SINGLETON),
       smallestEndpointIdentifier: x.SMALLEST_ENDPOINT_IDENTIFIER, // Smallest endpoint Id in which the attribute is present
+      storage: x.STORAGE_OPTION,
+      storageOption: x.STORAGE_OPTION,
+      tokenId: x.TOKEN_ID, // Endpoint type attribute's token id
+      type: x.TYPE, // Attribute type
```

Comment:
Why does this not match the "attribute" map?  Is that because the underlying query it's used for does this "stick the ARRAY_TYPE in TYPE" bit?  And if so, why the difference in implementation?

Reply:
All the maps in db-mapping are actually an exact representation of the database tables. I wanted to keep that map the same way for now. I created an extended one because this does not just have entries from the endpointTypeAttribute table but other tables as well. You are also right about the array_type in type bit but that is not the reason I made a separate map.

This way we can bring all the maps floating around in the code into db-mapping.

Reply:
My point is this: all maps for attributes should have "type" meaning the same thing.  And right now we don't seem to be doing that.

Reply:
I forgot type was already set to `type: x.TYPE != 'array' ? x.TYPE : x.ARRAY_TYPE,` for attribute. Fixed this